massa-protocol-worker/src/protocol_worker.rs

```diff
@@ -1127,6 +1143,23 @@ impl ProtocolWorker {
         }
 
         if !new_operations.is_empty() {
+            for (op_id, _) in received_ids.iter() {
+                if let Some(block_id) = self.awaiting_operations.get(op_id) {
+                    if let Some(info) = self.checked_headers.get_mut(block_id) {
+                        info.awaiting_operations.remove(op_id);
+                        if info.awaiting_operations.is_empty() {
+                            self.awaiting_operations.remove(op_id);
```

Comment:
I don't understand why you remove the operation id only if the block has all his dependency filled. Because you inserted all operations of the block on this vector.

For example if your block have 10 operations and you received the first 9 you will not remove any from `self. awaiting_operations` and then remove only the 10th

Reply:
Ah yes that's an error, thanks for pointing it out.